File: Ping/cpp/ping.cpp
/*
 * IPWorks 2022 C++ Edition - Demo Application
 *
 * Copyright (c) 2023 /n software inc. - All rights reserved. - www.nsoftware.com
 *
 */


#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include "../../include/ipworks.h"
#define LINE_LEN 100

int main(int argc, char *argv[])
{

	Ping ping;
	int ret_code;
	int i;

	if (argc != 2)
	{
		fprintf(stderr, "usage:  ping hostname\n");
		fprintf(stderr, "\n");
		fprintf(stderr, "  hostname the IP address or Domain Name of the remote host\n");
		fprintf(stderr, "\nExample ping www.yahoo.com\n\n");
		printf("Press enter to continue.");
		getchar();
	}
	if(ret_code = ping.SetRemoteHost(argv[1]) )  goto done;

	ping.SetPacketSize(32);
	printf("\nPinging %s with %i bytes of data:\n\n", argv[1], ping.GetPacketSize());

	for (i=0; i<4; i++)
	{
		ping.SetTimeToLive(255);
		if (ret_code = ping.PingHost(argv[1])) goto done;
		printf("Reply from %s: bytes=%i time=%ims\n", ping.GetRemoteHost(), ping.GetPacketSize(), ping.GetResponseTime());
	}

done:
	if (ret_code)     // Got an error.  The user is done.
	{
		printf( "\nError: %d", ret_code );
		if (ping.GetLastError())
		{
			printf( " \"%s\"\n", ping.GetLastError() );
		}
	}
	printf("Press enter to continue.");
	getchar();  
	return 0;
}

File: Ping/python/ping.py
#
# IPWorks 2022 Python Edition - Demo Application
#
# Copyright (c) 2023 /n software inc. - All rights reserved. - www.nsoftware.com
#

import sys
import string
from ipworks import *

input = sys.hexversion<0x03000000 and raw_input or input


if len(sys.argv) != 2:
  print("usage: ping.py host")
  print("")
  print("  host  the IP address (IP number in dotted internet format) or Domain Name of the remote host")
  print("\r\nExample: ping.py www.google.com")
else:
  ping = Ping()

  try:
    ping.set_remote_host(sys.argv[1])
    print("\nPinging %s.\n" %(sys.argv[1]))
    for i in range(1,5):
      ping.set_timeout(15)
      ping.ping_host(sys.argv[1])
      if ping.get_response_time() == 0:
        print("Reply from %s: bytes=%i time=0ms" %(ping.get_remote_host(),ping.get_packet_size()))
      else:
        print("Reply from %s: bytes=%i time=%sms" %(ping.get_remote_host(),ping.get_packet_size(),ping.get_response_time()))

  except IPWorksError as e:
    print("ERROR %s" %e.message)



